index.js

```diff
@@ -1039,6 +1039,66 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  //# takeWhile :: (Foldable f, Monoid (f a)) => (a -> Boolean) -> f a -> f a
+  //.
+  //. Discards the first inner value which does not satisfy the predicate, and
+  //. all subsequent inner values.
+  //.
+  //. ```javascript
+  //. > S.takeWhile(S.odd, [3, 3, 3, 7, 6, 3, 5, 4])
+  //. [3, 3, 3, 7]
+  //.
+  //. > S.takeWhile(S.even, [3, 3, 3, 7, 6, 3, 5, 4])
+  //. []
+  //. ```
+  function takeWhile(pred, xs) {
+    if (Array.isArray(xs)) {
+      for (var idx = 0, result = [], x;
+           idx < xs.length && pred(x = xs[idx]);
+           idx += 1) result.push(x);
+      return result;
+    }
```

Comment:
You'd prefer `Array$takeWhile` to a generic `findIndex`? The definition of `Array$takeWhile` and `Array$dropWhile` will be nearly identical save the arguments to `slice`. We don't have to expose it.